content/components/pages/NamespacesPage.tsx

```diff
@@ -0,0 +1,252 @@
+import EnabledTag from "devtools/components/recipes/EnabledTag";
+import NamespacePicker from "devtools/components/recipes/NamespacePicker";
+import RecipeCard from "devtools/components/recipes/RecipeCard";
+import { NAMESPACES } from "devtools/config";
+import {
+  useSelectedEnvironmentState,
+  useSelectedNormandyEnvironmentAPI,
+} from "devtools/contexts/environment";
+import {
+  NamespacesProvider,
+  useNamespaceInfo,
+} from "devtools/contexts/namespaces";
+import { RecipeV3 } from "devtools/types/recipes";
+import { getSamplingFilterAsNamespaceSample } from "devtools/utils/recipes";
+
+// default export
+const NamespacesPage: React.FC = () => {
+  const normandyApi = useSelectedNormandyEnvironmentAPI();
+  return (
+    <NamespacesProvider normandyApi={normandyApi}>
+      <PageWrapper>
+        <NamespaceListing />
+      </PageWrapper>
+    </NamespacesProvider>
+  );
+};
+
+export default NamespacesPage;
+
+const NamespaceListing: React.FC = () => {
+  const { selectedKey: selectedEnvironment } = useSelectedEnvironmentState();
+  const [selectedNamespace, setSelectedNamespace] = useState(
+    NAMESPACES.default,
+  );
+  const namespacesInfo = useNamespaceInfo();
+
+  const namespaces = namespacesInfo.namespaces;
+  namespaces.add(selectedNamespace);
+
+  const handleNamespaceChange = useCallback(
+    (value) => {
+      setSelectedNamespace(value);
+    },
+    [setSelectedNamespace],
+  );
+
+  const fullBucketCount = namespacesInfo
+    .findOccupiedBuckets(selectedNamespace)
+    .filter((occupied) => occupied).length;
+
+  useEffect(() => {
+    namespacesInfo.updateNamespace(selectedNamespace);
+  }, [selectedNamespace, selectedEnvironment]);
+
+  const recipeLists = categorizeRecipes(
+    namespacesInfo.recipesByNamespace[selectedNamespace],
+  );
+
+  return (
+    <>
+      <div className="d-flex align-items-center">
+        <h3>Recipes in Namespace</h3>
+        <NamespacePicker
+          className="ml-1"
+          value={selectedNamespace}
+          onChange={handleNamespaceChange}
+        />
+        {!namespacesInfo.isLoading("updateNamespace", selectedNamespace) && (
+          <div className="ml-1">{fullBucketCount / 100}% full</div>
+        )}
+      </div>
+
+      {recipeLists.ok.length === 0 &&
+        !namespacesInfo.isLoading("updateNamespace", selectedNamespace) && (
+          <p>
+            No{" "}
+            {(!!recipeLists.backwards.length || !!recipeLists.error.length) &&
+              "standard formatted"}{" "}
+            recipes found.
+          </p>
+        )}
+      {namespacesInfo.isLoading("updateNamespace", selectedNamespace) ? (
+        <div className="grid-layout grid-3 card-grid mt-2">
+          <CardPlaceholder />
+          <CardPlaceholder />
+          <CardPlaceholder />
+        </div>
+      ) : (
+        <div className="grid-layout grid-3 card-grid mt-2">
+          {recipeLists.ok.map(({ recipe, bucketRange }) => (
```

Comment:
i find these long ternary situations kind of hard to read. i always prefer something like:

```js
let something = <div>stuff here</div>;
if (condition) {
  something = <div>other stuff</div>;
}

...

{something}
```